src/unix/bsd/mod.rs

```diff
@@ -10,6 +10,11 @@ pub type sa_family_t = u8;
 pub type pthread_t = ::uintptr_t;
 pub type nfds_t = ::c_uint;
 
+#[cfg(not(target_os = "dragonfly"))]
+macro_rules! utsname_len { () => (256) }
+#[cfg(target_os = "dragonfly")]
+macro_rules! utsname_len { () => (32) }
```

Comment:
This may also be better done with a private `const` as that won't get reexported up. Could you also put this next to the definition of `utsname`?